### PR TITLE
Fix user registry migration logic + print out registry when asked

### PR DESCRIPTION
**Pull request by cewert**
## Changes
- Fix bug with user migration logic
- Utilize `bs_const` conditional vars to print out the registry when enabled. defaults to disabled.
  - Prints first thing before the app does anything
  - Prints again after any registry migrations have been ran

### CODE REVIEW

File: source/utils/conditional.bs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+' @fileoverview Conditional functions that depend on 'bs_const' values specified in the manifest file
+
+' Print out all of the registry contents to the debug log
+sub printRegistry()
+    #if printReg
```
Comment (1hitsong) [inline source/utils/conditional.bs:5]: Do we have the option to use use `not` to exit early and avoid the long if statement?

`#if not printReg then return`

File: source/migrations.bs [hunk 2]
```diff
@@ -44,17 +45,27 @@ sub runGlobalMigrations()
...
+            reg = CreateObject("roRegistrySection", section)
+            if reg.exists("LastRunVersion")
+                lastRunVersion = reg.read("LastRunVersion")
+            else
+                ' fall back to using the apps lastRunVersion
+                ' NOTE: app versions < 2.0.0 didn't save this at the user level
+                lastRunVersion = m.global.app.lastRunVersion
+            end if
+
+            if not versionChecker(lastRunVersion, "2.0.0")
```
Comment (1hitsong) [inline source/migrations.bs]: We should move the `2.0.0` value to a constant.

File: source/utils/conditional.bs (hunk 1 above)
Comment (cewert) [reply source/utils/conditional.bs:5]: Don't think so

not-conditional

File: source/migrations.bs (hunk 2 above)
Comment (cewert) [reply source/migrations.bs]: I took your name suggestion. I'm using a brighterscript constant and all caps with underscores for the styling

**Review by 1hitsong — APPROVED** [review]

**Pull request merged**